azure-iot-provisioning-devicesdk/tests/device/test_registration_client_factory.py

```diff
@@ -0,0 +1,63 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+from azure.iot.provisioning.devicesdk.security.sk_security_client import SymmetricKeySecurityClient
+from azure.iot.provisioning.devicesdk.transport.mqtt_transport import MQTTTransport
+from azure.iot.provisioning.devicesdk.device.sk_registration_client import (
+    SymmetricKeyRegistrationClient,
+)
+from azure.iot.provisioning.devicesdk.device.registration_client_factory import (
+    create_from_security_provider,
+)
+
+
+xfail_notimplemented = pytest.mark.xfail(raises=NotImplementedError, reason="Unimplemented")
+
+
+@pytest.fixture
+def provisioning_host():
+    return "fakehost.com"
+
+
+@pytest.fixture
+def security_client():
+    return SymmetricKeySecurityClient("fake_registration_id", "fake_symmetric_key", "fake_id_scope")
+
+
+@pytest.mark.parametrize(
+    "protocol,expected_transport",
```

Comment:
I'd also parametrize different types of security clients. Right now there's no test for matching protocols with security clients.

Reply:
right now since I have not introduced the other security clients I did not include that. as far as the transports are concerned we always know there are 3 types of transport.

---
In reply to: [270211088](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r270211088) [](ancestors = 270211088)